sdk/eventhub/Azure.Messaging.EventHubs/src/Core/DeveloperCodeException.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Messaging.EventHubs
+{
+    /// <summary>
+    ///   Serves an marker for an exception occurring within developer-provided code, such as
+    ///   event handlers.  Such exceptions are typically intended to be explicitly not handled by
+    ///   the infrastructure of the various Event Hubs types.
+    /// </summary>
+    ///
+    /// <seealso cref="System.Exception" />
+    ///
+    internal class DeveloperCodeException : Exception
```

Comment:
For context:  this exception is used within the processor infrastructure only to explicitly mark an exception that was observed in developer-provided code, such as an event handler.  It is not exposed to callers directly.